### PR TITLE
add ssl connection test

### PR DESCRIPTION
**Pull request by gkorland**

### CODE REVIEW

**Comment by gkorland** [comment]
@xianwill  
```
com.redislabs.provider.redis.util.ConnectionSSLUtilsTest *** ABORTED ***
  java.lang.RuntimeException: Unable to load a Suite class that was discovered in the runpath: com.redislabs.provider.redis.util.ConnectionSSLUtilsTest
  at org.scalatest.tools.DiscoverySuite$.getSuiteInstance(DiscoverySuite.scala:81)
  at org.scalatest.tools.DiscoverySuite$$anonfun$1.apply(DiscoverySuite.scala:38)
  at org.scalatest.tools.DiscoverySuite$$anonfun$1.apply(DiscoverySuite.scala:37)
  at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
  at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
  at scala.collection.Iterator$class.foreach(Iterator.scala:891)
  at scala.collection.AbstractIterator.foreach(Iterator.scala:1334)
  at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
  at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
  at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
  ...
  Cause: redis.clients.jedis.exceptions.JedisConnectionException: Could not get a resource from the pool
  at redis.clients.jedis.util.Pool.getResource(Pool.java:59)
  at redis.clients.jedis.JedisPool.getResource(JedisPool.java:234)
  at com.redislabs.provider.redis.ConnectionPool$.connect(ConnectionPool.scala:35)
  at com.redislabs.provider.redis.RedisEndpoint.connect(RedisConfig.scala:72)
  at com.redislabs.provider.redis.RedisConfig.clusterEnabled(RedisConfig.scala:196)
  at com.redislabs.provider.redis.RedisConfig.getNodes(RedisConfig.scala:320)
  at com.redislabs.provider.redis.RedisConfig.getHosts(RedisConfig.scala:236)
  at com.redislabs.provider.redis.RedisConfig.<init>(RedisConfig.scala:135)
  at com.redislabs.provider.redis.env.RedisStandaloneSSLEnv$class.$init$(RedisStandaloneSSLEnv.scala:20)
  at com.redislabs.provider.redis.util.ConnectionSSLUtilsTest.<init>(ConnectionSSLUtilsTest.scala:13)
  ...
  Cause: redis.clients.jedis.exceptions.JedisConnectionException: javax.net.ssl.SSLException: Connection has been shutdown: javax.net.ssl.SSLHandshakeException: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
  at redis.clients.jedis.Connection.disconnect(Connection.java:222)
  at redis.clients.jedis.BinaryClient.disconnect(BinaryClient.java:918)
  at redis.clients.jedis.Connection.close(Connection.java:212)
  at redis.clients.jedis.BinaryClient.close(BinaryClient.java:924)
  at redis.clients.jedis.BinaryJedis.close(BinaryJedis.java:1934)
  at redis.clients.jedis.Jedis.close(Jedis.java:3508)
  at redis.clients.jedis.JedisFactory.makeObject(JedisFactory.java:128)
  at org.apache.commons.pool2.impl.GenericObjectPool.create(GenericObjectPool.java:819)
  at org.apache.commons.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:429)
  at org.apache.commons.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:360)
  ...
  Cause: javax.net.ssl.SSLException: Connection has been shutdown: javax.net.ssl.SSLHandshakeException: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
  at sun.security.ssl.SSLSocketImpl.checkEOF(SSLSocketImpl.java:1533)
  at sun.security.ssl.SSLSocketImpl.checkWrite(SSLSocketImpl.java:1545)
  at sun.security.ssl.AppOutputStream.write(AppOutputStream.java:71)
  at redis.clients.jedis.util.RedisOutputStream.flushBuffer(RedisOutputStream.java:52)
  at redis.clients.jedis.util.RedisOutputStream.flush(RedisOutputStream.java:133)
  at redis.clients.jedis.Connection.disconnect(Connection.java:218)
  at redis.clients.jedis.BinaryClient.disconnect(BinaryClient.java:918)
  at redis.clients.jedis.Connection.close(Connection.java:212)
  at redis.clients.jedis.BinaryClient.close(BinaryClient.java:924)
  at redis.clients.jedis.BinaryJedis.close(BinaryJedis.java:1934)
  ...
  Cause: javax.net.ssl.SSLHandshakeException: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
  at sun.security.ssl.Alerts.getSSLException(Alerts.java:192)
  at sun.security.ssl.SSLSocketImpl.fatal(SSLSocketImpl.java:1946)
  at sun.security.ssl.Handshaker.fatalSE(Handshaker.java:316)
  at sun.security.ssl.Handshaker.fatalSE(Handshaker.java:310)
  at sun.security.ssl.ClientHandshaker.serverCertificate(ClientHandshaker.java:1639)
  at sun.security.ssl.ClientHandshaker.processMessage(ClientHandshaker.java:223)
  at sun.security.ssl.Handshaker.processLoop(Handshaker.java:1037)
  at sun.security.ssl.Handshaker.process_record(Handshaker.java:965)
  at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:1064)
  at sun.security.ssl.SSLSocketImpl.performInitialHandshake(SSLSocketImpl.java:1367)
  ...
  Cause: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
  at sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:397)
  at sun.security.validator.PKIXValidator.engineValidate(PKIXValidator.java:302)
  at sun.security.validator.Validator.validate(Validator.java:262)
  at sun.security.ssl.X509TrustManagerImpl.validate(X509TrustManagerImpl.java:324)
  at sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:229)
  at sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(X509TrustManagerImpl.java:124)
  at sun.security.ssl.ClientHandshaker.serverCertificate(ClientHandshaker.java:1621)
  at sun.security.ssl.ClientHandshaker.processMessage(ClientHandshaker.java:223)
  at sun.security.ssl.Handshaker.processLoop(Handshaker.java:1037)
  at sun.security.ssl.Handshaker.process_record(Handshaker.java:965)
  ...
  Cause: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
  at sun.security.provider.certpath.SunCertPathBuilder.build(SunCertPathBuilder.java:141)
  at sun.security.provider.certpath.SunCertPathBuilder.engineBuild(SunCertPathBuilder.java:126)
  at java.security.cert.CertPathBuilder.build(CertPathBuilder.java:280)
  at sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:392)
  at sun.security.validator.PKIXValidator.engineValidate(PKIXValidator.java:302)
  at sun.security.validator.Validator.validate(Validator.java:262)
  at sun.security.ssl.X509TrustManagerImpl.validate(X509TrustManagerImpl.java:324)
  at sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:229)
  at sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(X509TrustManagerImpl.java:124)
  at sun.security.ssl.ClientHandshaker.serverCertificate(ClientHandshaker.java:1621)
```

**Comment by xianwill** [comment]
@gkorland - thanks so much for adding this test suite. I did my testing as an integration test against a TLS only AWS Elasticache instance. Looking at this failure, I think for broader support against Redis instances that may be hosted anywhere signed by any CA cert - I probably need to add an additional hook to allow inclusion of the CA cert. I can take a closer look at this tonight.

File: Makefile [hunk 1]
```diff
@@ -9,6 +9,23 @@ appendonly no
 requirepass passwd
 endef
 
+# STANDALONE REDIS NODE WITH SSL
```
Comment (fe2s) [inline Makefile:12]: Travis tests are not running in this branch for some reason.

When I tried to run it locally I'm getting the following

```
Oleksiis-MacBook-Pro:Projects oleksiidiagiliev$ cd spark-redis
Oleksiis-MacBook-Pro:spark-redis oleksiidiagiliev$ make start
make start-standalone
echo "$REDIS_STANDALONE_NODE_CONF" | redis-server -
echo "$REDIS_STANDALONE_NODE_CONF_SSL" | redis-server -

*** FATAL CONFIG FILE ERROR (Redis 6.0.1) ***
Reading the configuration file, at line 8
>>> 'tls-auth-clients no'
Bad directive or wrong number of arguments
make[1]: *** [start-standalone] Error 1
make: *** [start] Error 2
```

File: Makefile [hunk 2]
```diff
@@ -98,7 +118,7 @@ test:
 
 benchmark:
 	make start
-	mvn clean test -B -Pbenchmark
+	mvn clean test -B -Pbenchmark -DargLine="-Djavax.net.ssl.trustStorePassword=password -Djavax.net.ssl.trustStore=/home/guy/redisclients/spark-redis/src/test/resources/tls/clientkeystore -Djavax.net.ssl.trustStoreType=jceks"
```
[the remark this answers is not in the document]
Comment (gkorland) [reply Makefile]: :+1:

**Comment by gkorland** [comment]
@xianwill found how to set the keystore, the pom.xml for the tests

**Pull request merged**